addons/web_editor/static/src/components/media_dialog/file_selector.js

```diff
@@ -288,7 +288,13 @@ export class FileSelector extends Component {
         await fetch(url).then(async result => {
             const blob = await result.blob();
             blob.id = new Date().getTime();
-            blob.name = new URL(url).pathname.split("/").findLast(s => s);
+            let docHref = this.props.media.ownerDocument.defaultView.location.href;
+            if (docHref === "about:srcdoc") {
+                // If the iframe has a `srcdoc` attribute, use the url of the
+                // main window as a base for the URL creation.
+                docHref = window.location.href;
```

Comment:
The goal of this it to take [this fix](https://github.com/odoo/odoo/commit/f56e2f4212d7fe3ac5e65e1ff45fe4f356589ad3) into account